src/1_utils.js

```diff
@@ -553,3 +553,49 @@ utils.getBrowserLanguageCode = function() {
 	}
 	return code;
 };
+
+/**
+ * Returns the difference between two arrays.
+ */
+utils.calculateDiffBetweenArrays = function(original, toCheck) {
+	var diff = [];
+	toCheck.forEach(function(element) {
+		if (original.indexOf(element) === -1) {
+			diff.push(element);
+		}
+	});
+	return diff;
+};
+
+/**
+ * Validates the commerce-data object passed into branch.trackCommerceEvent().
+ * If there are invalid keys present then it will report back what those keys are.
+ * Note: The keys below are optional.
+ */
+utils.validateCommerceData = function(commerceData) {
+	var allowedInRoot = [ 'common', 'type', 'transaction_id', 'currency', 'revenue', 'revenue_in_usd', 'exchange_rate', 'shipping', 'tax', 'coupon', 'affiliation', 'persona', 'products' ];
+	var allowedInProducts = [ 'sku', 'name', 'price', 'quantity', 'brand', 'category', 'variant' ];
+
+	var invalidKeys = {
+		root: utils.calculateDiffBetweenArrays(allowedInRoot, Object.keys(commerceData)),
+		products: []
+	};
+
+	if (commerceData.hasOwnProperty('products')) {
+		commerceData["products"].forEach(function(product) {
+			invalidKeys['products'] = invalidKeys['products'].concat(utils.calculateDiffBetweenArrays(allowedInProducts, Object.keys(product)));
+		});
```

Comment:
I don't think this would work since commereData["products"] is an array of objects. You would be comparing the list from `allowedInProducts` which is an array of strings